src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -433,39 +444,40 @@ public synchronized void unsubscribe(UnsubscribeRequest request)
         return CompletableFuture.completedFuture(0);
     }
 
-    private synchronized void spoolMessage()  {
-        if (spoolingFuture.get() == null || spoolingFuture.get().isCancelled()) {
-            spoolingFuture.set(ses.scheduleWithFixedDelay(this::spoolTask, 0, 5, TimeUnit.SECONDS));
+    protected CompletableFuture<Integer> spoolSingleMessage() throws InterruptedException {
+        final long id = spool.popId();
```

Comment:
does this call block until an ID is available, or what happens if the spooler is empty?

Reply:
If the spooler is empty, it will pause here and wait until there is new message inserted into the queue.